tempesta_fw/sock_clnt.c

```diff
@@ -348,6 +357,83 @@ tfw_listen_sock_stop_all(void)
 }
 
 /*
+ * Check, if client addr matches server addr
+ * server - backend address:port
+ * listener - listener address:port, address can be 0.0.0.0
+ */
+int
+tfw_sock_match_server_client(const TfwAddr *server, const TfwAddr *listener)
+{
```

Comment:
The function should be in addr.[ch]. Also there is tfw_addr_eq() and standard IN_LOOPBACK() which should be used to minimize code of the function.